akeneo-design-system/src/components/Input/TextInput/TextInput.tsx

```
@@ -0,0 +1,116 @@
+  width: 100%;
+  height: 40px;
+  border: 1px solid ${({invalid}) => (invalid ? getColor('red', 100) : getColor('grey', 80))};
+  border-radius: 2px;
+  background: ${({readOnly}) => (readOnly ? getColor('grey', 20) : getColor('white'))};
+  color: ${getColor('grey', 140)};
+  font-size: ${getFontSize('default')};
+  line-height: 40px;
+  padding: 0 15px;
+  box-sizing: border-box;
+  outline-style: none;
+
+  &:focus {
+    box-shadow: 0 0 0 2px ${getColor('blue', 40)};
+  }
+
+  &::placeholder {
+    color: ${getColor('grey', 100)};
+  }
+`;
+
+const ReadOnlyIcon = styled(LockIcon)`
+  position: absolute;
+  right: 0;
+  top: 0;
+  margin: 12px;
+  color: ${getColor('grey', 100)};
+`;
+
+const CharacterLeftLabel = styled.div`
+  font-size: ${getFontSize('small')};
+  align-self: flex-end;
+  color: ${getColor('grey', 100)};
+`;
+
+type TextInputProps = Override<
+  React.InputHTMLAttributes<HTMLInputElement> & InputProps,
+  {
+    /**
+     * Value of the input.
+     */
+    value: string;
+
+    /**
+     * Placeholder displayed when the input is empty.
+     */
+    placeholder?: string;
+
+    /**
+     * Defines if the input should be read only. If defined so, the user cannot change the value of the input.
+     */
+    readOnly?: boolean;
+
+    /**
+     * Defines if the input is valid on not.
+     */
+    invalid?: boolean;
+
+    /**
+     * Label displayed under the field to display the remaining character counter.
+     */
+    characterLeftLabel?: string;
+
+    /**
+     * Handler called when the value of the input changes.
+     */
+    onChange?: (newValue: string) => void;
+  }
+>;
+
+/**
+ * The TextInput component allows the user to enter content and data when the expected input is a single line of text.
+ */
+const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
+  ({invalid, onChange, readOnly, characterLeftLabel, ...rest}: TextInputProps, forwardedRef: Ref<HTMLInputElement>) => {
+    const handleChange = useCallback(
```

Should we throw an Error when input is not read only and there is no onChange ? Maybe we can enforce it by the same way of BooleanProps (https://github.com/akeneo/pim-community-dev/pull/13525/files?short_path=0c9323d#diff-ea18ef73ba4f8ee3df3744190e9dd750bdaf3b7c4214fd2f660b724b39caf946R102)

---

Enforced by type here https://github.com/akeneo/pim-community-dev/pull/13525/commits/0f4a0bdd1cf30dde04bebbf7b4732a367c05f183